src/devices/Ft232H/Ftx232HDevice.cs

```diff
@@ -0,0 +1,1074 @@
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Iot.Device.Ft232H;
+using Iot.Device.Board;
+using Iot.Device.Ft2232H;
+using Iot.Device.Ft4232H;
+
+namespace Iot.Device.FtCommon
+{
+    /// <summary>
+    /// FTx32H base Device
+    /// </summary>
+    public class Ftx232HDevice : FtDevice, IDisposable
+    {
+        private const int PinNumberFT2x = 16;
+        private const int PinNumberFT4x = 8;
+
+        /// <summary>
+        /// Number of pins
+        /// </summary>
+        internal const int PinCountConst = 16;
+
+        // Commands are available in:
+        // - AN_108_Command_Processor_for_MPSSE_and_MCU_Host_Bus_Emulation_Modes.pdf
+        // - AN_113_FTDI_Hi_Speed_USB_To_I2C_Example.pdf
+        // To understand the signal need for I2C communication: https://training.ti.com/sites/default/files/docs/slides-i2c-protocol.pdf
+        private const uint I2cMasterFrequencyKbps = 400;
+        // Direction values
+        private const byte I2cDirSDAinSCLin = 0x00;
+        private const byte I2cDirSDAinSCLout = 0x01;
+        private const byte I2cDirSDAoutSCLin = 0x02;
+        private const byte I2cDirSDAoutSCLout = 0x03;
+        // Data values
+        private const byte I2cDataSDAloSCLhi = 0x01;
+        private const byte I2cDataSDAhiSCLhi = 0x03;
+        private const byte I2cDataSDAloSCLlo = 0x00;
+        private const byte I2cDataSDAhiSCLlo = 0x02;
+        private const byte NumberCycles = 6;
+        private const byte MaskGpio = 0xF8;
+
+        private SafeFtHandle _ftHandle = null!;
+
+        // This is used by FT232H and others to track the GPIO states
+        internal byte GpioLowData = 0;
+        internal byte GpioLowDir = 0;
+        internal byte GpioHighData = 0;
+        internal byte GpioHighDir = 0;
+
+        internal bool[] PinOpen = new bool[PinCountConst];
+        internal PinMode[] GpioDirections = new PinMode[PinCountConst];
+        internal bool UsesMpseeForGpio = true;
+
+        internal List<SpiConnectionSettings> ConnectionSettings = new List<SpiConnectionSettings>();
+
+        /// <summary>
+        /// Gets all the FT232H connected
+        /// </summary>
+        /// <returns>A list of FT232H</returns>
+        public static List<Ftx232HDevice> GetFtx232H()
+        {
+            List<Ftx232HDevice> ft232s = new List<Ftx232HDevice>();
+            var devices = FtCommon.GetDevices(new FtDeviceType[] { FtDeviceType.Ft232H, FtDeviceType.Ft2232H, FtDeviceType.Ft4232H, FtDeviceType.Ft2232 });
+            foreach (var device in devices)
+            {
+                var dev = device.Type switch
+                {
+                    FtDeviceType.Ft2232 or FtDeviceType.Ft2232H => new Ft2232HDevice(device),
+                    FtDeviceType.Ft232H => new Ft232HDevice(device),
+                    FtDeviceType.Ft4232H => new Ft4232HDevice(device),
+                    _ => new Ftx232HDevice(device),
```

Comment:
Is this fallback useful? Are you assuming that any unknown device will just work with the fallback?

Reply:
yes, see the sample where I'm using the Ftx232HDevice rather than a named one.

Reply:
ho, got your question, you're right, it should not be needed but I leave it like this otherwise the switch complains that there is no default